### ISSUE TITLE
RabbitMQ helmchart ignores NodePort values when type is set to LoadBalancer

### ISSUE THREAD

**Issue by devedse** [bitnami/charts#8363]
**Which chart**:
rabbitmq: 3.9.11

**Describe the bug**
When deploying RabbitMQ with `.Values.service.type == "LoadBalancer"` it's not possible to configure the NodePort. This causes Kubernetes to randomly assign a NodePort which (in our case) conflicted with other hardcoded NodePorts.

This happens for multiple ports, for example the `distPort` in `svc.yaml`
```
    - name: {{ .Values.service.distPortName }}
      port: {{ .Values.service.distPort }}
      targetPort: dist
      {{- if eq .Values.service.type "ClusterIP" }}
      nodePort: null
      {{- else if (and (eq .Values.service.type "NodePort") (not (empty .Values.service.distNodePort))) }}
      nodePort: {{ .Values.service.distNodePort }}
```

It would be nice to add an `or` condition here that allows to provide a custom NodePort when the `.Values.service.type` is set to `LoadBalancer` or `NodePort`.

The issue also happens for the ports: `dist`, `epmd` and `http`.

**To Reproduce**
When setting `type` to `LoadBalancer` even if you specify the NodePort it still results in random NodePorts being assigned:

Deployment 1:
![image](https://user-images.githubusercontent.com/2350015/145557993-4ec9b70f-2552-45c7-8760-6429f9049d63.png)

Deployment 2:
![image](https://user-images.githubusercontent.com/2350015/145558021-852c1a56-694d-49b3-9798-dc7d2fac7c62.png)

**Expected behavior**
It would be nice to be able to specify NodePorts when the type is set to either NodePort or LoadBalancer.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.2.4", GitCommit:"0ad800ef43d3b826f31a5ad8dfbb4fe05d143688", GitTreeState:"clean", GoVersion:"go1.13.12"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5", GitCommit:"aea7bbadd2fc0cd689de94a54e5b7b758869d691", GitTreeState:"clean", BuildDate:"2021-09-15T21:10:45Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5", GitCommit:"aea7bbadd2fc0cd689de94a54e5b7b758869d691", GitTreeState:"clean", BuildDate:"2021-09-15T21:04:16Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
N/A

**Comment by juan131** [comment]
Hi @devedse 

> This causes Kubernetes to randomly assign a NodePort which (in our case) conflicted with other hardcoded NodePorts.

How's that possible? Kubernetes is supposed to assign a random port (within the `30000-32767` range by default) but excluding those already assigned to existing services. 

> It would be nice to be able to specify NodePorts when the type is set to either NodePort or LoadBalancer.

I don't see any problem on allowing that. Indeed we're setting the `nodePort` based on the `.Values.service.nodePort` both when the service is NodePort or LoadBalancer, see:

- https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/templates/svc.yaml#L42

**Comment by devedse** [comment]
Hi @juan131 , thanks for your quick response. The issue happens because we're mixing hardcoded ports and randomly set ports.
If we first deploy a HelmChart with a random node port which gets assigned for example 30300. And afterwards we deploy our own application which has that same 30300 port assigned hard coded then this issue happens.

So we either need to set all NodePorts randomly or hardcoded. For now our approach is HardCoded since all of them except RabbitMQ are hardcoded already. We might re-evaluate that decision later.

Regarding your second point, currently for the `amqp` port his is indeed the case. It is however not the case for the following ports:
1. epmd
2. dist
3. stats

These are currently missing the `eq .Values.service.type "NodePort"`

**Comment by devedse** [comment]
I just created this PR to fix the issue:
https://github.com/bitnami/charts/pull/8369

**Comment by juan131** [comment]
> The issue happens because we're mixing hardcoded ports and randomly set ports.

That makes sense! Thanks for the clarification!

> I just created this PR to fix the issue: #8369

Thanks for the fix! @devedse 
During the next few hours a new chart release including your fix will be released.
